src/app/[locale]/(default)/rules/[subset]/page.tsx

```diff
@@ -0,0 +1,15 @@
+import { rulesMockdata } from '@/mock-data/rules';
+
+export default function SubPage({
+  params: { subset },
+}: { params: { subset: string } }) {
+  const page = rulesMockdata.find(
+    (rule) => rule.id === Number.parseInt(subset),
+  );
+  if (!page) return null;
```

Comment:
Instead of returning an empty page, please use `notFound` to return a 404 page instead